Sources/SwiftParser/StringLiterals.swift

```diff
@@ -475,11 +475,15 @@ extension Parser {
     var segments: [RawStringLiteralSegmentsSyntax.Element] = []
     var loopProgress = LoopProgressCondition()
     while loopProgress.evaluate(self.currentToken) {
+      // If we encounter a token with leading trivia, we're no longer in the
+      // string literal.
+      guard currentToken.leadingTriviaText.isEmpty else { break }
```

Comment:
Technically the condition here is that the leading trivia contains a newline, right? You are just using the implication here that `leadingTrivia.isEmpty` is only `true` if it contains a newline. I’d prefer to check for a newline here, just to be safe should we change the trivia attribution rule. So, wouldn’t `!currentToken.isAtStartOfLine` be better?

Reply:
> Technically the condition here is that the leading trivia contains a newline, right?

Yes, but we shouldn't accept any other form of trivia either, as string literals can't contain trivia outside of interpolations. I'm happy to change this to just check for a newline, though I think in that case we should then assert that we don't have any other kind of trivia. Does that sound reasonable?

> just to be safe should we change the trivia attribution rule

Could you elaborate on what you mean by this?

Reply:
> Yes, but we shouldn't accept any other form of trivia either, as string literals can't contain trivia outside of interpolations. I'm happy to change this to just check for a newline, though I think in that case we should then assert that we don't have any other kind of trivia. Does that sound reasonable?

Ah, yes that makes sense. Thanks for the explanation.

> Could you elaborate on what you mean by this?

Currently the lexer only puts newlines in leading trivia, never in trailing trivia, so non-empty leading trivia implies that there is a newline inside the trivia. But with your explanation above this doesn’t matter.